elements/pfe-primary-detail/src/pfe-primary-detail.js

```diff
@@ -0,0 +1,177 @@
+
+    // Add appropriate markup and behaviors
+    this._scanLightDom();
+    // Set first item as active for initial load
+    this._handleHideShow({ target: this._slots.detailsNav[0] });
+  }
+
+  // @todo Add mutation observer in case any content changes, rerun scanLightDom?
+
+  disconnectedCallback() {
+    for (let index = 0; index < this._slots.detailsNav.length; index++) {
+      this._slots.detailsNav[index].removeEventListener("click", this._handleHideShow);
+    }
+  }
+
+  /**
+   * Create nav functionality and adds additional HTML/attributes to markup
+   */
+  _scanLightDom() {
+    if (this._slots.detailsNav.length !== this._slots.details.length) {
+      console.error(
+        `${this.tag}: The number of item headings does not match the number of item details. Found ${this._slots.detailsNav.length} item headings & ${this._slots.details.length} item details.`
+      );
+      return;
+    }
+
+    // Setup left sidebar navigation
+    this._slots.detailsNav.forEach((toggle, index) => {
+      let attr = toggle.attributes;
+      toggle.dataset.index = index;
+
+      const button = document.createElement("button");
+
+      button.innerText = toggle.textContent;
+      // Copy over attributes from original element
+      [...attr].forEach(toggle => {
+        button.setAttribute(toggle.name, toggle.value);
+      });
+
+      // If the toggle does not have a ID, set a unique ID
+      if (!toggle.hasAttribute("id")) {
+        button.setAttribute(
+          "id",
+          `pfe-detail-toggle-${Math.random()
+            .toString(36)
+            .substr(2, 9)}`
+        );
+      }
+
+      button.addEventListener("click", this._handleHideShow);
+      this._slots.detailsNav[index] = button;
+      toggle.replaceWith(button);
+    });
+
+    // Setup item detail elements
+    this._slots.details.forEach((detail, index) => {
+      // If the toggle does not have a ID, set a unique ID
+      if (!detail.hasAttribute("id")) {
+        detail.setAttribute(
+          "id",
+          `pfe-detail-${Math.random()
+            .toString(36)
+            .substr(2, 9)}`
+        );
+      }
+
+      const toggleId = this._slots.detailsNav[index].getAttribute("id");
+      if (!detail.hasAttribute("aria-labelledby") && toggleId) {
+        detail.setAttribute("aria-labelledby", toggleId);
+      }
+
+      detail.dataset.index = index;
+    });
+
+    // Swing back to detailsNav to add aria-controls, now that details have an Id
+    this._slots.detailsNav.forEach((toggle, index) => {
+      const detailId = this._slots.details[index].getAttribute("id");
+      if (!toggle.hasAttribute("aria-controls") && detailId) {
+        toggle.setAttribute("aria-controls", detailId);
+      }
+    });
+  }
+
+  /**
+   * Handles changes in state
+   * @param {object} e Event object
+   */
+  _handleHideShow(e) {
+    if (!e.target.classList.contains("pfe-primary-detail__toggle--active")) {
```

Comment:
Since your active state corresponds with the aria-expanded state, you could just use that for styling instead of making a new class?  One less step.

Reply:
I've had issues with visibility and animation corresponding to aria attributes in the past, I'd prefer not to conflate them and keep aria functional.

Reply:
What kind of issues?  If you're setting the class at the same time as the aria, I'm not sure what issues we'd run into.  I'd love to hear more about why that would be problematic.

Reply:
Right, in this case at the moment there's no need, and come to think of it probably won't ever be. I can simplify it.